client/pages/google/index.tsx

```diff
@@ -303,14 +376,27 @@ const GoogleLoginPage = () => {
                   <p className="text-[#2eb150]">사용 가능한 전화번호입니다.</p>
                 )}
               </Stack>
-              <Button
-                variant="text"
-                className="rounded"
-                onClick={handleCheckPhoneNumber}
-                size="small"
-              >
-                중복 확인
-              </Button>
+              {!phoneNumberPreChecked && (
+                <Button
+                  variant="text"
+                  className="rounded"
+                  onClick={handleCheckPhoneNumber}
+                  size="small"
+                  disabled
+                >
+                  중복 확인
+                </Button>
+              )}
+              {phoneNumberPreChecked && (
+                <Button
+                  variant="text"
+                  className="rounded"
+                  onClick={handleCheckPhoneNumber}
+                  size="small"
+                >
+                  중복 확인
+                </Button>
+              )}
```

Comment:
컨디션별로 분기되는 컴포넌트를 확인해보면 disabled attribute 외에는 동일하므로 해당 attribute만 분기해주어도 될 것 같아요.
```suggestion
                <Button
                  variant="text"
                  className="rounded"
                  onClick={handleCheckPhoneNumber}
                  size="small"
                  disabled={!phoneNumberPreChecked}
                >
                  중복 확인
                </Button>
```

Reply:
ㅠㅠ 코드가 20줄은 줄었습니다 감사합니다

Reply:
오 disabled 분기를 이렇게도 할 수 있었네요 저는 {...phoneNumberPreChecked ? {disabled : true} : null } 이런식으로 했었는데 너무 좋은 것 같습니다